### ISSUE TITLE
Installing C dependencies

### ISSUE THREAD

**Issue by andreaferretti** [nim-lang/nimble#330]
Is there a way to install C libraries using nimble? Ideally, I would like to be able to

* write some nimble task that compiles a C shared library `libfoo.so`
* have `libfoo.so` copied inside some standard path (e.g. `.nimble/pkgs/mylib-myversion/lib/libfoo.so` or something)
* have nimble add the relevant path to LDLIBRARY when `mylib` is among the dependencies

Any suggestions how to do so?

**Comment by dom96** [comment]
> Is there a way to install C libraries using nimble?

Afraid not, and solving this isn't easy. As far as Nimble is concerned the DLLs need to be installed separately via a system package manager or manually by the user.

**Comment by andreaferretti** [comment]
Why isn't it easy? It would amount to launching a custom task after install (to compile the lib) and then the only thing nimble would need to do is automatically add a `clibdir` flag when compiling something that depends on the library, much like it adds to the standard nim path

**Comment by dom96** [comment]
Okay, so maybe it's not that difficult. But why not solve this differently: by always statically compiling?

**Comment by andreaferretti** [comment]
Because I have to link a CUDA library :-)

**Comment by genotrance** [comment]
I was able to pull this off with nimgen in `before install` but that has been broken for a while since nimble doesn't include any files generated in `before install` since the file scan happens before that. This is since 0.9.0. I will work on fixing that at some point.

https://github.com/nim-lang/nimble/issues/533

In nimterop, I have been able to pull and build C libs (tree-sitter) during [compile](https://github.com/nimterop/nimterop/blob/master/nimterop/setup.nim) time. It isn't saving anything into the `~/.nimble/pkgs/` directory due to #533 but at least external source can be consumed as part of a binary build.

This could probably be closed and tracked as part of #533.
